types/src/bytesrepr.rs

```diff
@@ -1317,6 +1087,58 @@ where
     }
 }
 
+/// Serializes dynamic bytes array.
+pub fn serialize_bytes(bytes: &[u8]) -> Result<Vec<u8>, Error> {
+    let serialized_length = bytes_serialized_length(bytes);
+    let mut vec = try_vec_with_capacity(serialized_length)?;
+    let length_prefix = bytes.len() as u32;
+    let length_prefix_bytes = length_prefix.to_le_bytes();
+    vec.extend(&length_prefix_bytes);
+    vec.extend(bytes);
+    Ok(vec)
+}
+
+/// Estimates serialized length of serialized bytes.
+pub fn bytes_serialized_length(bytes: &[u8]) -> usize {
+    U32_SERIALIZED_LENGTH + bytes.len()
+}
+
+/// Deserializes dynamic bytes array.
+pub fn deserialize_bytes(bytes: &[u8]) -> Result<(Vec<u8>, &[u8]), Error> {
+    let (size, remainder) = u32::from_bytes(bytes)?;
+    let (result, remainder) = safe_split_at(remainder, size as usize)?;
+    Ok((result.to_vec(), remainder))
+}
+
+/// Serializes array.
+pub fn serialize_array(bytes: &[u8]) -> Result<Vec<u8>, Error> {
```

Comment:
I think we shouldn't take a slice here.  I'd prefer:
```
pub fn u8_array_32_to_bytes(bytes: &[u8; 32]) -> Result<Vec<u8>, Error> {
```

Seems like we only _need_ to handle `[u8; 32]` and `[u8; 33]` right now, but it'd maybe be good to use a macro to generate handlers for the other array sizes we cover.